autoarray/structures/grids.py

```diff
@@ -762,11 +764,188 @@ def blurring_grid_from_mask_and_kernel_shape(
                 grid_1d=blurring_grid_1d
             )
 
+    def structure_from_result(self, result):
```

Comment:
Type annotations would be nice

Reply:
Given that it's multiple possible types you can use typing.Union

Reply:
Donoe.